web/modules/custom/dpl_po/src/Commands/DplPoCommands.php

```diff
@@ -0,0 +1,432 @@
+<?php
+
+namespace Drupal\dpl_po\Commands;
+
+use Drupal\Component\Gettext\PoHeader;
+use Drupal\Component\Gettext\PoStreamReader;
+use Drupal\Component\Gettext\PoStreamWriter;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\dpl_po\Services\CtpConfigManager;
+use Drush\Commands\DrushCommands;
+
+use function Safe\filemtime;
+use function Safe\preg_match;
+use function Safe\rename;
+
+/**
+ * A Drush commandfile.
+ */
+class DplPoCommands extends DrushCommands {
+  use StringTranslationTrait;
+
+  /**
+   * The source.
+   */
+  protected string $source;
+  /**
+   * The destination.
+   */
+  protected string $destination;
+  /**
+   * The language code of the proccessed po file.
+   */
+  protected string $languageCode;
+
+  /**
+   * Set the destination.
+   */
+  protected function setDestination(string $path): void {
+    $this->destination = $path;
+  }
+
+  /**
+   * Get the destination.
+   */
+  protected function getDestination(): ?string {
+    return $this->destination;
+  }
+
+  /**
+   * Set the source.
+   */
+  protected function setSource(string $path): void {
+    $this->source = $path;
+  }
+
+  /**
+   * Get the source.
+   */
+  protected function getSource(): ?string {
+    return $this->source;
+  }
+
+  /**
+   * Set the language code.
+   */
+  protected function setLanguageCode(string $langcode): void {
+    $this->languageCode = $langcode;
+  }
+
+  /**
+   * Get the language code.
+   */
+  protected function getLanguageCode(): ?string {
+    return $this->languageCode;
+  }
+
+  /**
+   * Class constructor.
+   */
+  public function __construct(
+    protected ConfigFactoryInterface $configFactory,
+    protected CtpConfigManager $cteiConfigManager,
+    protected FileSystemInterface $fileSystem,
+    protected ModuleHandlerInterface $moduleHandler
+  ) {}
+
+  /**
+   * Create a .po file with only the configuration strings.
+   *
+   * @param string $langcode
+   *   The langcode to import. Eg. 'en' or 'fr'.
+   * @param string $source
+   *   The path to the source .po file.
+   * @param string $destination
+   *   The path to the destination .po file.
+   *
+   * @command dpl_po:extract-config
+   * @usage drush dpl_po:extract-config da da.po
+   *   Extracts strings with config context and writes a fie with it.
+   */
+  public function createPoFileConfigOnly($langcode, $source, $destination): void {
+    $this->setLanguageCode($langcode);
+    $this->setSource($source);
+    $this->setDestination($destination);
+    $this->validateSource();
+    $this->validateDestination();
+
+    try {
+      $file = $this->extractTranslationsIntoFile('/^([a-z]+\.)+/');
```

Comment:
Please add a comment explaining why this pattern makes sense in indentifying configuration-related translations.

Since it is reused it make sense to move it to a class constant and document it there.

Reply:
yez